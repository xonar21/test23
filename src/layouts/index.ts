export * from "./auth";
export * from "./default";
