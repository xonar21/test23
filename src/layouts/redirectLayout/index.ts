export { default as RedirectLayout } from "./RedirectLayout";
