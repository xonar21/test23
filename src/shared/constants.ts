export const ERROR_MESSAGES = {
  API_MIDDLEWARE_PROMISE_NOT_FOUND: "Could not find promise in the HTTP Client. Make sure you passed the correct path.",
};

export const LocalStorageKeys = {
  ThemeMode: "THEME_MODE",
};

export const CookieKeys = {
  AuthToken: "AUTH_TOKEN",
};

export const QueryKeys = {
  ReturnUrl: "returnUrl",
};
