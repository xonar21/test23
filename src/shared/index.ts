export * from "./constants";
export * from "./utils";
export * from "./routes";
export * from "./test-ids";
export * from "./locale";
