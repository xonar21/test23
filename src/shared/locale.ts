import { i18n } from "next-i18next";

export enum Locale {
  Romanian = "ro",
  Russian = "ru",
}

export const translate = (key: string | string[]) => {
  return i18n?.t(key) as string;
};
