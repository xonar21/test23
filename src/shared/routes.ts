import { EntityType, UserPermissions } from "~/security";

export interface IApplicationRoute {
  path: string;
  authorized?: boolean;
  annonymous?: boolean;
  permissions?: Partial<UserPermissions>;
  avoidBreadcrumbRender?: boolean;
}

export interface IApplicationRoutes {
  Home: IApplicationRoute;
  Layout: IApplicationRoute;
  Posts: IApplicationRoute;
  Login: IApplicationRoute;
  Register: IApplicationRoute;
}

export const routes: IApplicationRoutes = {
  Home: {
    path: "/",
  },
  Layout: {
    path: "/layout",
  },
  Posts: {
    path: "/posts",
    authorized: true,
    permissions: { read: [EntityType.Post] },
  },
  Login: {
    path: "/auth/login",
    annonymous: true,
  },
  Register: {
    path: "/auth/register",
    annonymous: true,
  },
};
