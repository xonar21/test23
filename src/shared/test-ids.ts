export const testIds = {
  components: {
    authForm: {
      email: "auth-form-email-input",
      password: "auth-form-password-input",
      submitButton: "auth-form-submit-button",
    },
    header: {
      root: "header",
      buttons: {
        themeToggler: "header-theme-toggle-button",
        drawerToggler: "header-drawer-toggle-button",
        localeMenu: "header-locale-menu",
        authButton: "header-auth-button",
      },
      localeItem: "header-locale-item",
    },
    navigationDrawer: {
      root: "navigation-drawer",
      navigationLink: "navigation-link",
    },
    dialogs: {
      postFormDialog: {
        owner: "post-form-dialog-owner",
        tags: "post-form-dialog-tags",
        submitButton: "post-form-dialog-submit-button",
      },
      promptDialog: {
        confirmButton: "prompt-dialog-confirm-button",
        rejectButton: "prompt-dialog-reject-button",
      },
    },
  },
};
