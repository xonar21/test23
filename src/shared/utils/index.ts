export * from "./array-utils";
export * from "./date-utils";
export * from "./object-utils";
export * from "./routing-utils";
export * from "./string-utils";
export * from "./testing-utils";
