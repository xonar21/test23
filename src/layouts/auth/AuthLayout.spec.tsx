import { render } from "~/shared";
import { AuthLayout } from "~/layouts";

const mockUseMediaQuery = jest.fn(() => false);

jest.mock("@mui/material", () => ({
  ...jest.requireActual("@mui/material"),
  __esModule: true,
  useMediaQuery: () => mockUseMediaQuery(),
}));

describe("Auth Layout", () => {
  const layout = () => (
    <AuthLayout>
      <div>Content</div>
    </AuthLayout>
  );

  it("renders without crashing", () => {
    const { asFragment } = render(layout());
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    mockUseMediaQuery.mockImplementation(() => true);
    const { asFragment } = render(layout());
    expect(asFragment()).toMatchSnapshot();
  });
});
