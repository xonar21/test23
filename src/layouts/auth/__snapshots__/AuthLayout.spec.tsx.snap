// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Layout renders in dark mode 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-b1a79c"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiCard-root css-1gid1ep-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-jbbmo6"
      >
        <button
          aria-label="dark-mode-toggle"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-v52rcf-MuiButtonBase-root-MuiIconButton-root"
          data-testid="header-theme-toggle-button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WbSunnyIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Auth Layout renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1b96vmi"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiCard-root css-ptl33d-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-jbbmo6"
      >
        <button
          aria-label="dark-mode-toggle"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          data-testid="header-theme-toggle-button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="Brightness3Icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        Content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
